ttnn/cpp/ttnn/operations/reduction.hpp

```diff
@@ -163,9 +164,28 @@ struct Reduce {
     }
 };
 
+struct ExecuteArgMax {
+    static inline const std::array<TensorSchema, 1> input_tensor_schemas() {
+        return {ttnn::TensorSchema{
+            4, 4, {ttnn::bfloat16}, {ttnn::ROW_MAJOR_LAYOUT}, true, false, false, false}};
+    }
+
+    template <typename... Args>
+    static auto input_tensors_to_validate(const Tensor& input_tensor, Args&&... args) {
+        return std::forward_as_tuple(input_tensor);
+    }
+
+    static ttnn::Tensor execute_on_worker_thread(
```

Comment:
I recommend to pass `optional output_tensor` and another overload with `queue_id` in this PR. 
operation::run supports those arguments already so it is just a few lines.

Reply:
will do soon